### PR TITLE
setup frontend

### PR DESCRIPTION
**Pull request by emilyhfdong**
**Changes**

- setup react app
- setup redux 
- create hook for websocket connection
- handle websocket events 
- created shared button & card component

### CODE REVIEW

File: frontend/src/utils/websocket-hook.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  BackendWebsocketActions,
+} from "src/types"
+import { config } from "./environment"
+import useWebSocket from "react-use-websocket"
+
+export const useWSConnection = () => {
+  const dispatch = useDispatch()
+
+  const { sendMessage } = useWebSocket(config.websocketUrl, {
+    share: true,
```
Comment (cindyhalim) [inline frontend/src/utils/websocket-hook.ts:16]: what does this do?

File: frontend/package.json [hunk 2]
```diff
@@ -1,11 +1,52 @@
...
-  "version": "1.0.0",
-  "description": "",
-  "main": "index.js",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "5.15.0",
+    "@testing-library/react": "11.2.7",
+    "@testing-library/user-event": "12.8.3",
+    "@types/jest": "26.0.24",
```
Comment (cindyhalim) [inline frontend/package.json]: should we move these @types to dev dependencies?

**Review by cindyhalim — APPROVED** [review]
ITS SO BEAUTIFUL

File: frontend/package.json (hunk 2 above)
Comment (emilyhfdong) [reply frontend/package.json]: ofc!!

**Pull request merged**